compiler/lexer.nim

```diff
@@ -330,13 +330,15 @@ template eatChar(L: var TLexer, t: var TToken) =
   inc(L.bufpos)
 
 proc getNumber(L: var TLexer, result: var TToken) =
-  proc matchUnderscoreChars(L: var TLexer, tok: var TToken, chars: set[char]) =
+  proc matchUnderscoreChars(L: var TLexer, tok: var TToken, chars: set[char]): uint =
```

Comment:
We don't use ``uint`` for counting, ``uint`` is for bit level hacking. Use ``Natural`` instead.